### DISCUSSION TITLE
Usage with models

### DISCUSSION THREAD

**Discussion by rrrrando** [spatie/laravel-data#139] [Q&A] [answered]
Hey!

I'm trying to understand how to use this package with models. Let's say i have a PostData object with relationships like this:

```php
$postData = PostData {
  +title: "Post title",
  +content: 'Post Content',
  +user: UserData {
    name: 'User name',
    email: 'User email'
  }
}
```

In the db it would be to tables connected with a foreign key.

If i now wanted to update the title and save it to the database, how would i do that? How to convert the DTO back to a model for updating?

**Comment by Nir-An** [comment, marked as answer]
@rrrrando  you can do something like [From a request](https://spatie.be/docs/laravel-data/v1/as-a-data-transfer-object/request-to-data-object) : 

call or bind the model and then insert what you want from the object 
```PHP

class SongController{
    ...
    
    public function update(
        Song $model, // the model
        SongData $songData //the  object
    ){
        $model->update(['title' => $songData->title]);
        
        return 
    }
}

```
for more information, you can see how to [update a model ](https://laravel.com/docs/9.x/eloquent#updates) and the difference between  'update' to 'save' method

**Reply by rrrrando** [reply]
Thanks! I kinda thought that might be the way. I was under the impression that whenever i need to modify some data i need to convert it to a dto, but thats not the case.
